doc/nrf/libraries/others/uart_async_adapter.rst

```diff
@@ -0,0 +1,104 @@
+.. _lib_uart_async_adapter:
+
+UART async adapter
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+UART async adapter library works as an interface between interrupt-driven API of a UART device and software that is using the asynchronous API interface.
+
+.. note::
+   The current implementation is :ref:`experimental <software_maturity>`.
+
+Overview
+********
+
+Zephyr provides the following three different ways to access a UART peripheral:
+
+* Pooling API
+* Interrupt-driven API
+* Asynchronous API
+
+While most of the UART drivers provide all three APIs, some drivers still do not provide the asynchronous UART API.
+An example of such a driver is USB CDC ACM.
+The UART async adapter library is initialized with a UART device and communicates with it using the interrupt-driven API, providing an asynchronous API for the application.
+
+This removes the need to implement two ways of interfacing with UART devices in software.
+When using the UART async adapter library, the software only needs to implement interaction with a UART device using the asynchronous API and just initialize the adapter for the ones that do not support it.
+
+Requirements
+************
+
+Enabling this library would also enable the :kconfig:option:`CONFIG_SERIAL_SUPPORT_ASYNC` Kconfig option.
+Without support for UART async API, this library does not even compile.
+You also need to enable the :kconfig:option:`CONFIG_SERIAL_SUPPORT_INTERRUPT` option.
+
+Configuration
+*************
+
+Use the :kconfig:option:`CONFIG_UART_ASYNC_ADAPTER` Kconfig option to enable the library in the build system.
+
+Usage
+*****
+
+See the following code snippet to learn how to use this library:
+
+.. code-block:: c
+
+    #include <uart_async_adapter.h>
+
+    ...
+    /* Get the UART device we want to use */
+    static const struct device *uart = DEVICE_DT_GET([UART device name]);
```

Comment:
The parameter for `DEVICE_DT_GET()` is not a UART device name but its devicetree node identifier.